### PR TITLE
Cross cluster consumer interest.

### PR DESCRIPTION
**Pull request by derekcollison**
In certain circumstances we were not taking gateway interest into consideration for consumers.
Also upgrade Go client and more raft stability fixes around split votes under heavy load.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/raft.go [hunk 1]
```diff
@@ -2090,7 +2104,7 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 			if eae, err := n.loadEntry(ae.pindex); err == nil && eae != nil {
 				// If terms mismatched, delete that entry and all others past it.
 				if ae.pterm > eae.pterm {
-					n.wal.Truncate(ae.pindex - 1)
+					n.wal.Truncate(ae.pindex)
```
Comment (kozlovic) [inline server/raft.go:2107]: again an off by one? :-)

**Review by kozlovic — APPROVED** [review]
LGTM

File: server/raft.go (hunk 1 above)
Comment (derekcollison) [reply server/raft.go:2107]: yea

**Review by matthiashanel — APPROVED** [review]
LGTM

**Pull request merged**